client/faucet/src/mint.rs

```diff
@@ -79,12 +85,28 @@ impl MintParams {
                 self.currency_code(),
                 self.receiver(),
                 self.amount,
-                vec![],
+                self.bcs_metadata(),
                 vec![],
             ),
         )
     }
 
+    fn bcs_metadata(&self) -> Vec<u8> {
+        let trade_id = match self.trade_id.clone() {
+            Some(trade_id) => trade_id,
+            _ => Uuid::new_v4().to_string(),
```

Comment:
Given a string is less compact than 128bits, maybe we should get some clarity on the exact intended format these are supposed to take onchain so we can figure out if it really should be a string or if we can compact this to just 128 bits.

Reply:
I think we should not create metadata when trade_id is not provided, otherwise it indicates this metadata is always used.

Reply:
ended up doing this specifically because (I believe) for minting this value should always be provided by the vasp